code-challenges.test.js

```diff
@@ -52,5 +151,39 @@ const numbersToAdd2 = [0, 7, -8, 12]
 const numbersToAdd3 = []
 // Expected output: []
 
+describe("sum", () => {
+    const numbersToAdd1 = [2, 4, 45, 9]
+    const numbersToAdd2 = [0, 7, -8, 12]
+    const numbersToAdd3 = []
+    it("takes in an array and returns an array of the accumulating sum", () => {
+        expect(sum(numbersToAdd1)).toEqual([2, 6, 51, 60])
+        expect(sum(numbersToAdd2)).toEqual([0, 7, -1, 11])
+        expect(sum(numbersToAdd3)).toEqual([])
+    })
+})
+
+// FAIL  ./code-challenges.test.js
+// sum
+//   ✕ takes in an array and returns an array of the accumulating sum (1 ms)
+
+// ● sum › takes in an array and returns an array of the accumulating sum
+
+//   ReferenceError: sum is not defined
 
 // b) Create the function that makes the test pass.
+// PSEUDO
+// input: [2, 4, 45, 9], [0, 7, -8, 12], []
+// parameter is an array
+// use conditionals
+// output: [2, 4, 45, 9], [0, 7, -8, 12], []
+
+const sum = (array) => {
+    let newArr = []
+    return array.map((value, index) => {
+        if (index > 0) {
+            return newArr = value + newArr
+        } else {
+            return newArr = value
+        }
+    })
+}
```

Comment:
I see where you are going with this but I think you can simplify the approach. It makes sense to use a variable outside the map scope, but since map returns an array I think the variable can serve a simpler purpose.